controller/internal/enforcer/nfqdatapath/nfq_windows.go

```diff
@@ -0,0 +1,165 @@
+		if err != nil {
+			parsedPacket.Print(packet.PacketFailureCreate, d.packetLogs)
+		} else if parsedPacket.IPProto() == packet.IPProtocolTCP {
+			if packetType == packet.PacketTypeNetwork {
+				tcpConn, processError = d.processNetworkTCPPackets(parsedPacket)
+			} else {
+				tcpConn, processError = d.processApplicationTCPPackets(parsedPacket)
+			}
+		} else if parsedPacket.IPProto() == packet.IPProtocolUDP {
+			// process udp packet
+			if packetType == packet.PacketTypeNetwork {
+				udpConn, processError = d.ProcessNetworkUDPPacket(parsedPacket)
+			} else {
+				udpConn, processError = d.ProcessApplicationUDPPacket(parsedPacket)
+			}
+		} else {
+			processError = fmt.Errorf("invalid ip protocol: %d", parsedPacket.IPProto())
+		}
+
+		if processError != nil {
+			if parsedPacket.IPProto() == packet.IPProtocolTCP {
+				d.collectTCPPacket(&debugpacketmessage{
+					Mark:    mark,
+					p:       parsedPacket,
+					tcpConn: tcpConn,
+					udpConn: nil,
+					err:     processError,
+					network: packetType == packet.PacketTypeNetwork,
+				})
+			} else if parsedPacket.IPProto() == packet.IPProtocolUDP {
+				d.collectUDPPacket(&debugpacketmessage{
+					Mark:    mark,
+					p:       parsedPacket,
+					tcpConn: nil,
+					udpConn: udpConn,
+					err:     processError,
+					network: packetType == packet.PacketTypeNetwork,
+				})
+			}
+			// drop packet by not forwarding it
+			return 0
+		}
+
+		// accept the (modified) packet by forwarding it
+		var modifiedPacketBytes []byte
+		if parsedPacket.IPProto() == packet.IPProtocolTCP {
+			modifiedPacketBytes = make([]byte, parsedPacket.IPTotalLen())
+			copyIndex := copy(modifiedPacketBytes, parsedPacket.GetBuffer(0))
+			copyIndex += copy(modifiedPacketBytes[copyIndex:], parsedPacket.GetTCPOptions())
+			copyIndex += copy(modifiedPacketBytes[copyIndex:], parsedPacket.GetTCPData())
+			packetInfo.PacketSize = uint32(copyIndex)
+		} else {
+			modifiedPacketBytes = parsedPacket.GetBuffer(0)
+			packetInfo.PacketSize = uint32(len(modifiedPacketBytes))
+		}
+
+		if parsedPacket.WindowsMetadata.(*afinetrawsocket.WindowsPacketMetadata).IgnoreFlow {
+			packetInfo.IgnoreFlow = 1
+		}
+		dllRet, _, err := frontman.PacketFilterForwardProc.Call(uintptr(unsafe.Pointer(&packetInfo)), uintptr(unsafe.Pointer(&modifiedPacketBytes[0])))
+		if dllRet == 0 {
+			zap.L().Error(fmt.Sprintf("%s failed: %v", frontman.PacketFilterForwardProc.Name, err))
+		}
+
+		if parsedPacket.IPProto() == packet.IPProtocolTCP {
+			d.collectTCPPacket(&debugpacketmessage{
+				Mark:    mark,
+				p:       parsedPacket,
+				tcpConn: tcpConn,
+				udpConn: nil,
+				err:     nil,
+				network: packetType == packet.PacketTypeNetwork,
+			})
+		} else if parsedPacket.IPProto() == packet.IPProtocolUDP {
+			d.collectUDPPacket(&debugpacketmessage{
+				Mark:    mark,
+				p:       parsedPacket,
+				tcpConn: nil,
+				udpConn: udpConn,
+				err:     nil,
+				network: packetType == packet.PacketTypeNetwork,
+			})
+		}
```

Comment:
This if-then-else seems to be repeated multiple times and it makes it hard to follow. Any chance we can consolidate the reporting steps in a function and avoid this.

Reply:
will simplify code (linux and windows) in another branch and/or after this merge